modules/dns_common/dns_common.rb

```diff
@@ -15,12 +17,20 @@ def resolver
       Resolv::DNS.new(:nameserver => @server)
     end
 
-    def dns_find key
-      if match = key.match(/(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})/)
-        resolver.getname(match[1..4].reverse.join(".")).to_s
-      else
-        resolver.getaddress(key).to_s
+    def dns_find key, type='auto'
+      if type == 'auto'
```

Comment:
Also, looking at dns_common/record_test.rb, it's worth breaking logic used for the type == 'auto' case into a separate class: corresponding tests won't require mocking then.